File: Contests/Codeforces/Solutions 2023/271A_Beautiful_Year.java
/**
 *  author:  josuerom
 *  created: 27/11/22 17:30:21
**/
import java.util.*;

public class BeautifulYear {
   public static void main(String[] args) {
      Scanner sc = new Scanner(System.in);
      int n = sc.nextInt();
      boolean flag = true;
      while (flag) {
         n++;
         HashSet<Integer> set = new HashSet<>();
         set.add(Integer.parseInt((n+"").charAt(0)+""));
         set.add(Integer.parseInt((n+"").charAt(1)+""));
         set.add(Integer.parseInt((n+"").charAt(2)+""));
         set.add(Integer.parseInt((n+"").charAt(3)+""));
         if (set.size() == 4) flag = false;
      }
      System.out.println(n);
   }
}

File: Training/Codeforces/2025/800/271A_Beautiful_Year.cpp
/**
 *  author:  josuerom
 *  created: 11/03/23 22:12:16
**/
#include <iostream>
#include <unordered_set>
#include <string>
using namespace std;

int main() {
   int n;
   std::cin >> n;
   bool flag = true;
   while (flag) {
      n++;
      unordered_set<int> set;
      string s = std::to_string(n);
      for (char c : s) {
        set.insert(c - '0');
      }
      if (set.size() == 4) {
         flag = false;
      }
   }
   cout << n << '\n';
   return 0;
}